packages/dev/src/main.ts

```diff
@@ -105,119 +129,90 @@ export class NetlifyDev {
+    this.#cleanupJobs = []
     this.#features = {
       blobs: options.blobs?.enabled !== false,
+      edgeFunctions: options.edgeFunctions?.enabled !== false,
       environmentVariables: options.environmentVariables?.enabled !== false,
       functions: options.functions?.enabled !== false,
       headers: options.headers?.enabled !== false,
       redirects: options.redirects?.enabled !== false,
       static: options.staticFiles?.enabled !== false,
     }
+    this.#functionsServePath = path.join(projectRoot, '.netlify', 'functions-serve')
+    this.#headersHandler = { handle: async (_request: Request, response: Response) => response }
     this.#logger = options.logger ?? globalThis.console
-    this.#projectRoot = options.projectRoot ?? process.cwd()
+    this.#server = options.serverAddress
+    this.#projectRoot = projectRoot
   }
 
   private async handleInEphemeralDirectory(request: Request, destPath: string) {
-    // Functions
-    const userFunctionsPath =
-      this.#config?.config.functionsDirectory ?? path.join(this.#projectRoot, 'netlify/functions')
-    const userFunctionsPathExists = await isDirectory(userFunctionsPath)
-    const functions = this.#features.functions
-      ? new FunctionsHandler({
-          config: this.#config,
-          destPath: destPath,
-          projectRoot: this.#projectRoot,
-          settings: {},
-          siteId: this.#siteID,
-          timeouts: {},
-          userFunctionsPath: userFunctionsPathExists ? userFunctionsPath : undefined,
-        })
-      : null
-
-    // Headers
-    const headers = this.#features.headers
-      ? new HeadersHandler({
-          configPath: this.#config?.configPath,
-          configHeaders: this.#config?.config.headers,
-          projectDir: this.#projectRoot,
-          publishDir: this.#config?.config.build.publish ?? undefined,
-          logger: this.#logger,
-        })
-      : { handle: async (_request: Request, response: Response) => response }
-
-    // Redirects
-    const redirects = this.#features.redirects
-      ? new RedirectsHandler({
-          configPath: this.#config?.configPath,
-          configRedirects: this.#config?.config.redirects,
-          jwtRoleClaim: '',
-          jwtSecret: '',
-          notFoundHandler,
-          projectDir: this.#projectRoot,
-        })
-      : null
-
-    // Static files
-    const staticFiles = this.#features.static
-      ? new StaticHandler({
-          directory: this.#config?.config.build.publish ?? this.#projectRoot,
-        })
-      : null
-
     // Try to match the request against the different steps in our request chain.
     //
     // https://docs.netlify.com/platform/request-chain/
 
     // 1. Check if the request matches a function.
-    const functionMatch = await functions?.match(request)
+    const edgeFunctionMatch = await this.#edgeFunctionsHandler?.handle(request.clone())
+    if (edgeFunctionMatch) {
+      return edgeFunctionMatch
+    }
```

Comment:
nit: this naming is inconsistent with the rest of the code
```suggestion
    const edgeFunctionResponse = await this.#edgeFunctionsHandler?.handle(request.clone())
    if (edgeFunctionResponse) {
      return edgeFunctionResponse
    }
```

Reply:
Applied.